projects/module-2/whats-cookin-part-two-agile.md

```diff
@@ -0,0 +1,206 @@
+
+| Description | URL | Method | Required Properties for Request | Sample Successful Response |
+| --- | --- | --- | --- | --- |
+| Get all users | http://localhost:3001/api/v1/users | GET | none | An array containing all users |
+| Get all ingredients | http://localhost:3001/api/v1/ingredients | GET | none | An array containing all ingredients |
+| Get all recipes | http://localhost:3001/api/v1/recipes | GET | none | An array containing all recipes |
+| Add recipe to cook for a user | http://localhost:3001/api/v1/usersRecipes | POST | { userID: `<number>`, recipeID: `<number>` } | {message: "Recipe # `<recipeID>` was added for User # `<userID>`" } |
+
+## Error Handling
+
+Make proper error handling for your users to ensure they get data and submit their POST requests successfully. An example is handling the case where they submit their data and an error message returns from the request. How would you show helpful feedback to a user?
+
+---
+
+## Iteration 5 - Differentiation Tracks
+
+Each team will get **one** of these tracks to work through.
+
+- The user wants to be able to decide what to make based on the cost of the recipe.
+- Users often print out recipes, but they don’t want to waste ink and paper. How can we give them a print layout with only the information they need from the recipe?
+- We’re getting a lot of international users these days, and they don’t know how much a dollar is exactly. We need to give them a way to convert the cost of a recipe into their own currency.
+- Sometimes a user doesn’t want to have to search through all the recipes when they come to the site. They want to see some featured recipes on the main page to help them make a quick decision.
+- Users want to plan a party using recipes from the app. After the party is planned, they need a grocery list to be able to make sure they have everything they need.
+- Give the users a way to remember if a recipe was good or not (this does not have to be saved on refresh, but it’s a nice bonus).
+- The admin of the site wants a way to see which recipes are getting the most hits (clicks). Give them some kind of tracking information that the admin can find.
+
+## Usability Testing
+
+As developers, we can get so accustomed to the features we create that we forget that users don’t have any background or know how to use a feature for the first time. That’s why it’s important to observe unbiased users try out your features and see where they succeed or have trouble. These observations are called usability tests.
+
+Once your track’s feature is at a point where it’s usable and ready for some feedback, but before you think everything is finalized, that’s the right time to perform the usability test.
+
+Your group will create a usability test for another group to go through. Here is the process:
+
+1. Ask someone from another project group to be the tester (participant)
+2. Schedule a time to do the usability test with the participant
+3. Create a script for the usability test
```

Comment:
Maybe there could be a note below these instructions re-emphasizing the importance of providing the participant with specific tasks, rather than saying "Just tell us what is tricky". Testing is an active, not passive, part of the development process.